src/middlewared/middlewared/plugins/account.py

```diff
@@ -663,11 +663,26 @@ async def __common_validation(self, verrors, data, schema, pk=None):
                         f'The path for the home directory "({data["home"]})" '
                         'must include a volume or dataset.'
                     )
+                elif os.path.realpath(data['home']) in [x['path'] for x in await self.middleware.call('pool.query')]:
+                    verrors.add(
+                        f'{schema}.home',
+                        f'The specified path is a ZFS pool mountpoint "({data["home"]})" '
+                    )
```

Comment:
We could probably get around without that extra expensive pool.query call?
Disallowing two level deep path?
/mnt/$anything?